tests/v2/validation/rbac/clusterandprojectroles/restrictedadmin_role_test.go

```diff
@@ -0,0 +1,117 @@
+package clusterandprojectroles
+
+import (
+	"testing"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	management "github.com/rancher/shepherd/clients/rancher/generated/management/v3"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/extensions/clusters/kubernetesversions"
+	"github.com/rancher/shepherd/extensions/provisioning"
+	"github.com/rancher/shepherd/extensions/provisioninginput"
+	rbac "github.com/rancher/shepherd/extensions/rbac"
+	"github.com/rancher/shepherd/extensions/settings"
+	"github.com/rancher/shepherd/pkg/config"
+	"github.com/rancher/shepherd/pkg/session"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type RestrictedAdminTestSuite struct {
+	suite.Suite
+	client  *rancher.Client
+	session *session.Session
+	cluster *management.Cluster
+}
+
+func (ra *RestrictedAdminTestSuite) TearDownSuite() {
+	ra.session.Cleanup()
+}
+
+func (ra *RestrictedAdminTestSuite) SetupSuite() {
+	ra.session = session.NewSession()
+
+	client, err := rancher.NewClient("", ra.session)
+	require.NoError(ra.T(), err)
+
+	ra.client = client
+
+	log.Info("Getting cluster name from the config file and append cluster details in pr")
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(ra.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(ra.client, clusterName)
+	require.NoError(ra.T(), err, "Error getting cluster ID")
+	ra.cluster, err = ra.client.Management.Cluster.ByID(clusterID)
+	require.NoError(ra.T(), err)
+}
+
+func (ra *RestrictedAdminTestSuite) TestRestrictedAdminCreateCluster() {
+	subSession := ra.session.NewSession()
+	defer subSession.Cleanup()
+
+	_, restrictedAdminClient, err := rbac.SetupUser(ra.client, rbac.RestrictedAdmin.String())
+	require.NoError(ra.T(), err)
+	ra.T().Logf("Validating restricted admin can create an RKE1 cluster")
+	userConfig := new(provisioninginput.Config)
+	config.LoadConfig(provisioninginput.ConfigurationFileKey, userConfig)
+	nodeProviders := userConfig.NodeProviders[0]
```

Comment:
There should be a readme for this, otherwise you'll run into user error when trying to run the test. Most of the time, we don't require dynamically configuration from the user for RBAC tests. Could this be statically defined instead?

Reply:
Sure we can define it statically too but the restricted admin is deprecated and will be removed sooner. so, we can keep these tests intact and I'll update the readme for now.